packages/docgen/test/get-type-annotation.js

```diff
@@ -47,237 +38,297 @@ describe( 'Type annotations', () => {
-			[ 'TSVoidKeyword', 'void' ],
+			'any',
+			'BigInt',
+			'boolean',
+			'never',
+			'null',
+			'number',
+			'object',
+			'string',
+			'symbol',
+			'undefined',
+			'unknown',
+			'void',
 		];
 
 		it.each( keywordTypes )(
 			`should get the parameter type for an %s`,
-			( paramType, expected ) => {
-				const node = getSimpleTypeNode( { paramType } );
-				expect( getTypeAnnotation( paramTag, node, 0 ) ).toBe(
-					expected
-				);
+			( typeName ) => {
+				expect(
+					getTypeAnnotation(
+						{ tag: 'param', name: 'foo' },
+						parse( `function fn(foo: ${ typeName }): any {}` ),
+						0
+					)
+				).toBe( typeName );
 			}
 		);
 
 		it.each( keywordTypes )(
 			`should get the return type for an %s`,
-			( returnType, expected ) => {
-				const node = getSimpleTypeNode( { returnType } );
-				expect( getTypeAnnotation( returnTag, node, 0 ) ).toBe(
-					expected
-				);
+			( typeName ) => {
+				expect(
+					getTypeAnnotation(
+						{ tag: 'return' },
+						parse( `function fn(foo: any): ${ typeName } {}` )
+					)
+				).toBe( typeName );
 			}
 		);
 	} );
 
 	describe( 'arrays, applied generic types, unions, intersections', () => {
-		const node = getArraysGenericTypesUnionsAndIntersctionsNode();
+		const node = parse( `
+			function fn( foo: MyType< string | number >[] ): MyType< string & number >[] {
+				return [];
+			}
+		` );
 
 		it( 'should get the param type', () => {
-			expect( getTypeAnnotation( paramTag, node, 0 ) ).toBe(
-				'MyType< string | number >[]'
-			);
+			expect(
+				getTypeAnnotation( { tag: 'param', name: 'foo' }, node, 0 )
+			).toBe( 'MyType< string | number >[]' );
 		} );
 
 		it( 'should get the return type', () => {
-			expect( getTypeAnnotation( returnTag, node, 0 ) ).toBe(
+			expect( getTypeAnnotation( { tag: 'return' }, node ) ).toBe(
 				'MyType< string & number >[]'
 			);
 		} );
 	} );
 
 	describe( 'literal values', () => {
-		describe.each( [
-			[ 'param', paramTag ],
-			[ 'return', returnTag ],
-		] )( '%s', ( _, tag ) => {
-			it.each( [
-				[ 'StringLiteral', 'a-string-literal', "'a-string-literal'" ],
-				[ 'BigIntLiteral', 1000, '1000n' ],
-				[ 'BooleanLiteral', true, 'true' ],
-				[ 'NumericLiteral', 1000, '1000' ],
-			] )(
-				'should handle %s',
-				( literalType, literalValue, literalResult ) => {
-					const node = getLiteralsNode( {
-						literalType,
-						literalValue,
-					} );
-					expect( getTypeAnnotation( tag, node, 0 ) ).toBe(
-						literalResult
-					);
+		it.each( [ "'a-string-literal'", '1000n', 'true', '1000' ] )(
+			'should handle %s',
+			( literal ) => {
+				const node = parse( `
+				function fn( foo: ${ literal } ): ${ literal } {
+					return BigInt( '1000' ) as ${ literal };
 				}
-			);
-		} );
+			` );
+				expect(
+					getTypeAnnotation( { tag: 'param', name: 'foo' }, node, 0 )
+				).toBe( literal );
+				expect( getTypeAnnotation( { tag: 'return' }, node ) ).toBe(
+					literal
+				);
```

Comment:
It'd be nice for these to be separate tests so they fail separate from each other (the code that handles them is relatively separate as well). It'd be annoying for this test to fail on the first assertion, fix it, only to find out that it was also failing on the second, mostly unrelated assertion.